template/src/components/Toast.tsx

```diff
@@ -1,202 +1,168 @@
-        Animated.delay(100),
+export const Toast: React.FC = () => {
+  const insets = useSafeAreaInsets()
+  const [state, setState] = useState<IToastState>(initState)
+  const HEIGHT = insets.bottom + metrics.toast
+  const animationRef = useRef<Animated.CompositeAnimation>()
+  const animation = useRef(new Animated.Value(0)).current
+  const offset = animation.interpolate({
+    inputRange: [0, 1],
+    outputRange: [HEIGHT, 0 - metrics.xxs], // padding bottom metrics.xxs
+  })
+  const opacity = animation.interpolate({
+    inputRange: [0, 1],
+    outputRange: [0, 1],
+  })
+
+  const displayMessage = (args: IToastState): void => {
+    animation.setValue(0)
+    setState(args)
+    animationRef.current?.stop()
+    setTimeout(() => {
+      animationRef.current = Animated.sequence([
         // Fade In
-        Animated.parallel([
-          Animated.timing(this.opacity, {
-            toValue: 1,
-            duration: 300,
-            useNativeDriver: true,
-          }),
-          Animated.timing(this.offset, {
-            toValue: 0,
-            duration: 300,
-            useNativeDriver: true,
-          }),
-        ]),
-        Animated.delay(3000),
-        // Fade Out
-        Animated.parallel([
-          Animated.timing(this.opacity, {
-            toValue: 0,
-            duration: 300,
-            useNativeDriver: true,
-          }),
-          Animated.timing(this.offset, {
-            toValue: HEIGHT * -1,
-            duration: 300,
-            useNativeDriver: true,
-          }),
-        ]),
-      ])
-      this.animated.start()
-    })
-  }
-
-  dismiss = () => {
-    this.animated?.stop()
-    Animated.parallel([
-      Animated.parallel([
-        Animated.timing(this.opacity, {
-          toValue: 0,
+        Animated.timing(animation, {
+          toValue: 1,
           duration: 300,
           useNativeDriver: true,
         }),
-        Animated.timing(this.offset, {
-          toValue: HEIGHT * -1,
+        Animated.delay(5000),
+        // Fade Out
+        Animated.timing(animation, {
+          toValue: 0,
           duration: 300,
           useNativeDriver: true,
         }),
-      ]),
-    ]).start()
+      ])
+      animationRef.current.start(() => {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+      })
+    }, 100)
   }
 
-  messageColor = (): string => {
-    const {type} = this.state
-
-    if (type === EToastType.SUCCESS) {
-      return colors.success
-    }
-    if (type === EToastType.INFO) {
-      return colors.info
-    }
-    return colors.error
+  const dismiss = () => {
+    animationRef.current?.stop()
+    setState(initState)
+    Animated.timing(animation, {
+      toValue: 0,
+      duration: 300,
+      useNativeDriver: true,
+    }).start()
   }
 
-  render() {
-    const {message, type} = this.state
-    return (
-      <Animated.View
+  useEffect(() => {
+    Emitter.on('SHOW_TOAST_MESSAGE', displayMessage)
```

Comment:
can u create constant for `SHOW_TOAST_MESSAGE` ?

Reply:
Added